eventlet/patcher.py

```diff
@@ -424,17 +424,160 @@ def _fix_py2_rlock(rlock, tid):
+        except KeyError:
+            pass
+        return "<%s %s.%s object owner=%r count=%d at %s>" % (
+            "locked" if self._block.locked() else "unlocked",
+            self.__class__.__module__,
+            self.__class__.__qualname__,
+            owner,
+            self._count,
+            hex(id(self))
+        )
+
+    def _at_fork_reinit(self):
+        self._block._at_fork_reinit()
+        self._owner = None
+        self._count = 0
+
+    def acquire(self, blocking=True, timeout=-1):
+        """Acquire a lock, blocking or non-blocking.
+
+        When invoked without arguments: if this thread already owns the lock,
+        increment the recursion level by one, and return immediately. Otherwise,
+        if another thread owns the lock, block until the lock is unlocked. Once
+        the lock is unlocked (not owned by any thread), then grab ownership, set
+        the recursion level to one, and return. If more than one thread is
+        blocked waiting until the lock is unlocked, only one at a time will be
+        able to grab ownership of the lock. There is no return value in this
+        case.
+
+        When invoked with the blocking argument set to true, do the same thing
+        as when called without arguments, and return true.
+
+        When invoked with the blocking argument set to false, do not block. If a
+        call without an argument would block, return false immediately;
+        otherwise, do the same thing as when called without arguments, and
+        return true.
+
+        When invoked with the floating-point timeout argument set to a positive
+        value, block for at most the number of seconds specified by timeout
+        and as long as the lock cannot be acquired.  Return true if the lock has
+        been acquired, false if the timeout has elapsed.
+
+        """
+        me = self._threading.get_ident()
+        if self._owner == me:
+            self._count += 1
+            return 1
+        rc = self._block.acquire(blocking, timeout)
+        if rc:
+            self._owner = me
+            self._count = 1
+        return rc
+
+    __enter__ = acquire
+
+    def release(self):
+        """Release a lock, decrementing the recursion level.
+
+        If after the decrement it is zero, reset the lock to unlocked (not owned
+        by any thread), and if any other threads are blocked waiting for the
+        lock to become unlocked, allow exactly one of them to proceed. If after
+        the decrement the recursion level is still nonzero, the lock remains
+        locked and owned by the calling thread.
+
+        Only call this method when the calling thread owns the lock. A
+        RuntimeError is raised if this method is called when the lock is
+        unlocked.
+
+        There is no return value.
+
+        """
+        # This breaks locks created by _fix_py3_rlock, so we comment it out:
+        #if self._owner != get_ident():
```

Comment:
From a functional perspective commenting this test seems a bit dangerous, isn't?
Without that we would release a un-acquired lock, I'm not an expert but this seems a non-sense to release something not aquired by the current process, and I don't think we want to do that.

However, as this test break py_rlock and as py_rlock is a forked buggy version of rlock, I'm not sure commenting this line will be worst than with this line un-commented.

Thoughts?

Reply:
Actually I agree with @4383 it seems a bit dangerous and in the case we want to have this line comment with need to add the space.

Reply:
is all python version impacted by this break, or simply recent version like 3.11?

May changes from https://github.com/eventlet/eventlet/pull/754/files could help to solve this problem and allow us to uncomment this condition. Especially the changes that modifying `eventlet/patcher.py` and the `fix_py3_rlock`.

However, to avoid pulling the entire ball of wool, we could leave this line commented for now (with the pep8 warning fixed), and, then, once merged, rewind on https://github.com/eventlet/eventlet/pull/754/ to see if we are now able to merge it too, and, then, try to uncomment this line once all the things are aligned.

Reply:
Versions previous to 3.11 have a slightly different _PyRLock implementation that doesn't have this line. So previous versions could just patch `threading.RLock` with `threading._PyRLock` and it "worked".